pkg/outdated/registry.go

```diff
@@ -9,14 +15,58 @@ import (
 	"github.com/pkg/errors"
 )
 
+type DockerConfig struct {
+	Auths map[string]DockerAuth `json:"auths"`
+}
+
+type DockerAuth struct {
+	Auth string `json:"auth"`
+}
+
 func initRegistryClient(hostname string) (*registry.Registry, error) {
 	if hostname == "docker.io" {
 		hostname = "index.docker.io"
 	}
 
+	username := ""
+	password := ""
+
+	if _, err := os.Stat(path.Join(homeDir(), ".docker", "config.json")); err == nil {
+		data, err := ioutil.ReadFile(path.Join(homeDir(), ".docker", "config.json"))
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to read docker config")
+		}
+
+		dockerConfig := DockerConfig{}
+		if err := json.Unmarshal(data, &dockerConfig); err != nil {
+			return nil, errors.Wrap(err, "failed to parse docker config")
+		}
+
+		for host, auth := range dockerConfig.Auths {
+			useAuth := false
+			if strings.Contains(host, "index.docker.io") && hostname == "index.docker.io" {
```

Comment:
How about just `strings.Contains(host, hostname)`.  And then you just need one if statement.  But even better is to parse the URL and get hostname from there.

Reply:
I'm not sure thats better. It would still work with docker.io, but if hostname is "quay.io" and the auth is for "myprivatequay.io" it would then be a false match?  the hostname for "docker.io" is a pretty special case in docker's auth file.